### PR TITLE
Processtx with OTP  [wip]

### PR DESCRIPTION
**Pull request by aeyakovenko**
just take a look at the src/otp.rs
does it make sense?

### CODE REVIEW

File: src/otp.rs [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+            let e = self.exit.lock().expect("lock");
+            if *e == true {
+                trace!("exiting");
+                return Ok(());
+            }
+        });
+        w.threads[port.into()] = Some(j);
+    }
+    pub fn listener<F>(&mut self, port: Port, func: F)
+        where F: (Data) -> Reply {
```
Comment (aeyakovenko) [inline src/otp.rs]: @garious so a simple service, takes  a message, returns a reply, which can forward it along.  why do this all?  so services are dead simple and dont have to manage the thread lifetimes, channels, etc...

Comment (garious) [reply src/otp.rs]: The agile approach is not to write code that isn't immediately useful. Looks like this particular function isn't even called.

Comment (aeyakovenko) [reply src/otp.rs]: It’s good to be called

**Comment by garious** [comment]
@aeyakovenko, can you find a more precise name for your OTP crate? Maybe add a comment that the implementation was inspired by Open Telecom Platform. If you're planning a OTP API compatibility, maybe a name tweak like ROTP (Rust OTP), otherwise a very different name.

Also, have you used `git rebase -i`? Multiple commits named only `wip` doesn't help me review the code. If the commits are standalone, then a descriptive commit message would be helpful. otherwise squash 'em.

Generally though, the code is looking good. Seems like the monolithic data.rs and result.rs should be split up and added to the modules with corresponding functionality. For result.rs, that might mean more `match` verbosity and less `?`, but I think it'll make the code more modular and those error cases more meaningful.

**Pull request closed without merging**